docs/dev-guide/design/extension-architecture.md

```
@@ -0,0 +1,195 @@
+# Optional fields
+version: string       # Semantic version
+author: string        # Author name or organization
+license: string       # SPDX license identifier
+```
+
+## Hello World: Bash Target Extension
+
+A minimal target extension that installs skills to a custom directory.
+
+**Directory structure:**
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)